typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,25 @@
+const $tabContainer = document.querySelector('.tab-container');
+if (!$tabContainer) throw new Error('$tabContainer not found!');
+$tabContainer.addEventListener('click', clickHandler);
+
+function clickHandler(event: Event): void {
+  const eventTarget = event.target as HTMLElement;
+  if (eventTarget.matches('.tab')) {
+    for (const tab of document.querySelectorAll('.tab')) {
```

Comment:
I would recommend a guard clause here to easier readability and to exit the function early if the `eventTarget` is not what we want.